### ISSUE TITLE
Give Options to Processor Preprocess

### ISSUE THREAD

**Issue by shadowtime2000** [eslint/eslint#13779]
**The version of ESLint you are using.**
7.11.0

**The problem you want to solve.**
Giving parser options to ESLint processors.

**Your take on the correct solution to problem.**
Allowing you to pass options to ESLint processors like this:

```javascript
module.exports = {
  overrides: [
    {
      files: ["*.foo"],
      processor: "foo/foo",
      options: {
        bar: false
      }
    }
  ]
};
```

**Are you willing to submit a pull request to implement this change?**
Yes

**Comment by shadowtime2000** [comment]
I am requesting a feature request so it is possible for users to pass configuration options to ESLint processors for ESLint plugins.

**Comment by mdjermanovic** [comment]
@shadowtime2000 is this the same as https://github.com/eslint/rfcs/tree/master/designs/2019-processor-shared-settings?

**Comment by shadowtime2000** [comment]
Yes.

**Comment by mdjermanovic** [comment]
Okay, the RFC I linked is already accepted and a PR is welcome :)

We had a PR https://github.com/eslint/eslint/pull/12068, but it wasn't finished. I'm not sure if anyone is already working on a new PR.
